radicle-node/src/service.rs

```diff
@@ -886,6 +879,34 @@ impl DerefMut for Sessions {
 
 mod gossip {
     use super::*;
+    use crate::service::filter::Filter;
+
+    #[derive(Default, Debug)]
+    pub struct Gossip {
+        received: Vec<(Timestamp, Announcement)>,
+    }
+
+    impl Gossip {
+        // TODO: Overwrite old messages from the same node or project.
+        // TODO: Should "time" be this node's time, or the time inside the message?
+        pub fn received(&mut self, ann: Announcement, time: Timestamp) {
+            self.received.push((time, ann));
+        }
+
+        pub fn filtered<'a>(
+            &'a self,
+            filter: &'a Filter,
+            start: Timestamp,
+            end: Timestamp,
+        ) -> impl Iterator<Item = Message> + '_ {
```

Comment:
Same as above re: action vs past tense.

Reply:
Same reasoning, we return a different type, whereas I'd expect a method called "filter" to return `Item = (Timestamp, Announcement)`